src/main/scala/org/ergoplatform/nodeView/wallet/scanning/Scan.scala

```diff
@@ -15,54 +18,72 @@ import scala.util.{Failure, Success, Try}
   * @param scanId         - unique identifier of an scan in the local system
   * @param scanName       - scan description (255 bytes in UTF-8 encoding max)
   * @param trackingRule  - a predicate to scan the blockchain for specific scan-related boxes
+  * @param walletInteraction - a flag which is prescribing how the scan is interacting with the p2pk-wallet
```

Comment:
Let's not call it a "flag" and use "type" or "kind" instead. Flag is usually binary (0 or 1), and here it is ternary and potentially even more.

Reply:
@aslesarenko flag is not necessarily a binary value:  https://techterms.com/definition/flag . 

Both terms you proposed are not so good here imho.

Reply:
I don't want to argue here, just a suggestion. 
From the definition "Flags are often binary flags" is kind of what I wrote above "Flag is usually binary".